src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import { selectTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useDispatch();
+  const todo = useSelector((state: RootState) => state.currentTodo);
+
+  // eslint-disable-next-line no-use-before-define
+  useEffect(() => {
+    setLoading(true);
+    getUser(todo.userId)
```

Comment:
There is a potential issue here if `todo` is null or undefined. Consider adding a check to ensure `todo` is not null before attempting to access `todo.userId` to prevent runtime errors.